recipes/opensubdiv/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+        "with_clew": [True, False],
+        "with_opencl": [True, False],
+        "with_dx": [True, False],
+        "with_metal": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "with_tbb": False,
+        "with_opengl": False,
+        "with_omp": False,
+        "with_cuda": False,
+        "with_clew": False,
+        "with_opencl": False,
+        "with_dx": False,
+        "with_metal": False,
+    }
+
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os != "Windows":
+            del self.options.with_dx
+        if self.settings.os != "Macos":
+            del self.options.with_metal
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            del self.options.shared
+
+    def requirements(self):
+        if self.options.with_tbb:
+            self.requires("onetbb/2020.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    @property
+    def _osd_gpu_enabled(self):
+        return any(
+            [
+                self.options.with_opengl,
+                self.options.with_opencl,
+                self.options.with_cuda,
+                self.options.get_safe("with_dx"),
+                self.options.get_safe("with_metal"),
+            ]
+        )
+
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["NO_TBB"] = not self.options.with_tbb
+        cmake.definitions["NO_OPENGL"] = not self.options.with_opengl
+        cmake.definitions["BUILD_SHARED_LIBS"] = self.options.get_safe("shared")
+        cmake.definitions["NO_OMP"] = not self.options.with_omp
+        cmake.definitions["NO_CUDA"] = not self.options.with_cuda
+        cmake.definitions["NO_DX"] = not self.options.get_safe("with_dx")
+        cmake.definitions["NO_METAL"] = not self.options.get_safe("with_metal")
+        cmake.definitions["NO_CLEW"] = not self.options.with_clew
+        cmake.definitions["NO_OPENCL"] = not self.options.with_opencl
+        cmake.definitions["NO_PTEX"] = True  # Note: PTEX is for examples only, but we skip them..
+        cmake.definitions["NO_DOC"] = True
+        cmake.definitions["NO_EXAMPLES"] = True
+        cmake.definitions["NO_TUTORIALS"] = True
+        cmake.definitions["NO_REGRESSION"] = True
+        cmake.definitions["NO_TESTS"] = True
+        cmake.definitions["NO_GLTESTS"] = True
+        if self.settings.os == "Linux" and self.settings.compiler == "clang" and not self.options.shared and self.settings.compiler.libcxx == "libc++":
+            cmake.definitions["CMAKE_CXX_FLAGS"] = "-stdlib=libc++"
+        if is_msvc(self) and self.settings.compiler.runtime in ["MT", "MTd"]:
```

Comment:
```suggestion
        if is_msvc_static_runtime(self):
```
https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc-static-runtime

don't forget to import